README.md

```diff
@@ -30,165 +32,471 @@ The two guiding principles are:
+
+#### Placeholders
+
+Hops supports a special syntax for placeholders. You can use angle brackets to reference other settings values:
+
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "basePath": "/foo",
+    "assetPath": "<basePath>/assets"
+  }
+}
+```
+
+In the above example the value of `<basePath>` inside `assetPath` will be replaced with `/foo`, so `assetPath` becomes `/foo/assets`.  
+In case of nested objects you can also use the dot notation to access nested values:
+
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "foo": {
+      "bar": "baz"
+    },
+    "qux": "<foo.bar>"
+  }
+}
 ```
 
-And write your main entry file (which should either be named `index.js` or should be referenced in your `package.json`s `main` field):
+#### Environment variables
+
+Hops has a concept of "universal environment variables", which are environment variables that are evaluated at _runtime_ instead of at _build time_.
+
+Usually when you reference `process.env.FOO` in your code, webpack will replace that expression with the value that the environment variable `FOO` had at build time. This is also true for Hops applications.
+
+**But:** Since Hops renders applications on the server- and client-side, we can reference environment variables in the settings and pass them on to the client and therefore allow users to have environment variables that are evaluated at runtime.
 
 **`package.json`**
 
 ```json
 {
-  "main": "main.js"
+  "hops": {
+    "myApiUrl": "[MY_API_URL]"
+  }
 }
 ```
 
-**`main.js`**
+The above example demonstrates how you can pass the value of the environment variable named `MY_API_URL` to Hops and later on you can access it in your application through its key like this:
+
+**`src/app.js`**
 
 ```javascript
-import { render } from 'hops-react';
+import { render, withConfig } from 'hops';
 import React from 'react';
 
-export default render(<h1>Hello World!</h1>);
+const Config = withConfig(({ config }) => <div>{config.myApiUrl}</div>);
+
+export default render(<Config />);
 ```
 
-For more details and advanced use-cases read the [full readme of `hops-react`](https://github.com/xing/hops/tree/master/packages/react).
+This allows you to have just one build artifact that you can use in different environments (staging, production, etc).
+
+**Caution:** Be careful not to expose secrets to the client side, because all environment variables that you reference this way in the settings will be visible in the rendered HTML.
+
+**Tip:** You can use Hops's configuration mechanism to provide custom values to your application. As seen in the environment variable example above, you can specify arbitrary keys in your settings and access them in your React application.
+
+### Options
 
-#### `hops-redux`
+There is another way to pass configuration to your Hops application, which is being used for runtime values:
```

Comment:
we should explicitly mention here that this is about passing an options object to the `render` function as a second argument. otherwise the readers will have to get this information from parsing the code block.